app/routes/application.js

```diff
@@ -4,4 +4,18 @@ import OSFAgnosticAuthRouteMixin from 'ember-osf/mixins/osf-agnostic-auth-route'
 import AnalyticsMixin from '../mixins/analytics-mixin';
 
 export default Ember.Route.extend(AnalyticsMixin, OSFAgnosticAuthRouteMixin, {
+    i18n: Ember.inject.service(),
+    afterModel: function() {
+        let locale;
+
+        // Works in Chrome and Firefox (editable in settings)
+        if (navigator.languages && navigator.languages[0])
+            locale = navigator.languages[0];
+        // Backup for Safari (uses system settings)
+        else if (navigator.language)
+            locale = navigator.language;
+
+        if (locale)
+            this.set('i18n.locale', locale);
```

Comment:
Yes, I tested it locally with different languages stubbed out and changing the settings on Chrome, Firefox, and Safari.

[The ember-i18n wiki documents this feature](https://github.com/jamesarosen/ember-i18n/wiki/Doc:-Setting-the-Locale/c2e2a1177ab7b6ac9e6c0e07301a164482d90855)